### PR TITLE
utils.match unit tests

### PR DESCRIPTION
**Pull request by aburrell**
Added more unit test for the routines in utils.match.

Partially addresses #6 

## Type of change

- Testing

# How Has This Been Tested?

This has been tested by running all the unit tests locally.

**Test Configuration**:
* Operating system: OSX Mojave
* Version number: 3.7.7

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by aburrell** [comment]
This is still a draft because I *still* can't get the pysat testing instruments to create a model instrument that can be interpolated over.  Looking for help.

File: pysatModels/tests/test_utils_match.py [hunk 1]
```diff
@@ -13,53 +16,102 @@
...
-        """ Runs before every method to create a clean testing setup."""
+        """ Runs before every method to create a clean testing setup
+        """
         self.ftime = dt.datetime(2009, 1, 1)
-        self.model_inst = pysat.Instrument(platform=str('pysat'),
-                                           name=str('testing_xarray'),
-                                           sat_id='12', clean_level='clean')
+        self.filename = "%Y-%m-%d.nofile"
+        self.model_kwargs = {'platform': str('pysat'),
+                             'name': str('testing_xarray'),
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_match.py:27]: This should be 'testmodel', not 'testing_xarray', which is an orbital object.

File: pysatModels/tests/test_utils_match.py [hunk 2]
```diff
@@ -13,53 +16,102 @@
...
         self.inst = pysat.Instrument(platform=str('pysat'),
                                      name=str('testing'),
                                      clean_level='clean')
         self.inst.load(yr=2009, doy=1)
         self.input_args = [dt.datetime(2009, 1, 1), dt.datetime(2009, 1, 2),
                            dt.timedelta(days=1), self.inst]
+        self.ref_col = self.inst.data.columns[0]
         self.model = pysat.Instrument(platform=str('pysat'),
-                                      name=str('testing_xarray'), sat_id='10',
+                                      name=str('testing2d_xarray'), sat_id='10',
```
Comment (jklenzing) [inline pysatModels/tests/test_utils_match.py:114]: Also 'testmodel'

**Comment by aburrell** [comment]
The different dimensions for each of the data caused a lot of headache, but I have the tests working now.

**Comment by aburrell** [comment]
Also, the fact that `testmodel` has altitude as a dimension, but the test instruments don't is a problem.

**Comment by aburrell** [comment]
Needs to have more coverage on the longitude handling.  The multi-dimensional testing will likely have to wait until the pysat test instruments improve.

**Comment by aburrell** [comment]
The only line not covered requires a change to the pysat instrument testing object.  Ready for review.

**Comment by aburrell** [comment]
Failures are flake8 changes.  @jklenzing, can you finish reviewing this branch before it gets even more out of sync?

**Comment by jklenzing** [comment]
In terms of line length issues, we can set flake8 to look at 80 characters, not 79.

**Comment by aburrell** [comment]
80 is what my editor is set to, obviously :P.  But the real issue is that these things have been fixed elsewhere, and this PR is getting stale.

**Pull request merged**